application-licensing-licensor/application-licensing-licensor-api/src/main/java/com/xwiki/licensing/internal/DefaultLicensingConfiguration.java

```diff
@@ -102,4 +110,34 @@ public File getLocalStorePath()
         }
     }
 
+    @Override
+    public String getStoreTrialURL()
+    {
+        return this.storeConfig.getProperty("storeTrialURL");
+    }
+
+    @Override
+    public String getStoreUpdateURL()
+    {
+        return this.storeConfig.getProperty("storeUpdateURL");
+    }
+
+    @Override
+    public String getLicensingOwnerFirstName()
+    {
+        return this.ownerConfig.getProperty("firstName", "");
```

Comment:
I'm not sure the empty string is useful as default value. Isn't that already the case if you don't specify a default value?

Reply:
The _null_ value is returned if no default value is given, leading to a NPE here https://github.com/xwikisas/application-licensing/pull/94/files/6261250232b15a50ff58bdb04459690455f09f15#diff-cfa828b50efb9640c7d353996f2d315d4074ef6b5835174fc0cdfc8560e6df08R199

Reply:
You could use ``StringUtils.isEmpty()`` there to avoid the NPE.